src/python/py/models/quantized_model.py

```diff
@@ -118,26 +126,25 @@ def __init__(self, quant_type, input_path, bits, group_size, q_size, kv_size, in
                         # Skip rotary embedding weights since they can be re-calculated when looping through the model
                         continue
                     elif name == "lm_head.qweight" or name == "transformer.output_layer.qweight":
-                        self._initialize_quantized_lm_head(bits, group_size)
+                        self._initialize_quantized_lm_head(local_bits, local_group_size)
                         self.lm_head.qweight = tensor
-                    elif name == "lm_head.qzeros" or name == "transformer.output_layer.qzeros":
-                        self._initialize_quantized_lm_head(bits, group_size)
+                    elif name in {"lm_head.qzeros", "lm_head.weight_zero_point", "transformer.output_layer.qzeros"}:
+                        self._initialize_quantized_lm_head(local_bits, local_group_size)
                         self.lm_head.qzeros = tensor
-                    elif name == "lm_head.scales" or name == "transformer.output_layer.scales":
-                        self._initialize_quantized_lm_head(bits, group_size)
+                    elif name in {"lm_head.scales", "lm_head.weight_scale", "transformer.output_layer.scales"}:
+                        self._initialize_quantized_lm_head(local_bits, local_group_size)
                         self.lm_head.scales = tensor
                     elif name == "lm_head.g_idx" or name == "transformer.output_layer.g_idx":
-                        self._initialize_quantized_lm_head(bits, group_size)
+                        self._initialize_quantized_lm_head(local_bits, local_group_size)
                         self.lm_head.g_idx = tensor
                     else:
                         if name.startswith("transformer.encoder"):
                             # Chatglm3, e.g., transformer.encoder.layers.0.input_layernorm.weight
                             name = name.replace("transformer.encoder", "model")
-                        curr_layer_id = int(name.split(".")[2])
-                        if curr_layer_id != layer_id:
-                            # Switch layer module used
-                            layer_id = curr_layer_id
-                            module = self.layers.setdefault(layer_id, QuantizedDecoderLayer(layer_id, bits, group_size))
+                        layer_id = int(name.split(".")[2])
+                        module = self.layers.setdefault(layer_id, QuantizedDecoderLayer(layer_id, local_bits, local_group_size))
```

Comment:
Moved module initialization to here. Since `module` is only used when `layer_id` is available. It is unused for above if-elif branches for other layers.